core/src/main/java/me/mykindos/betterpvp/core/combat/modifiers/impl/GenericModifier.java

```diff
@@ -0,0 +1,54 @@
+package me.mykindos.betterpvp.core.combat.modifiers.impl;
+
+
+import lombok.AllArgsConstructor;
+import lombok.With;
+import me.mykindos.betterpvp.core.combat.events.DamageEvent;
+import me.mykindos.betterpvp.core.combat.modifiers.DamageModifier;
+import me.mykindos.betterpvp.core.combat.modifiers.ModifierResult;
+import me.mykindos.betterpvp.core.combat.modifiers.ModifierType;
+
+/**
+ * Modifier that applies environmental damage effects based on location, weather, etc.
+ */
+@AllArgsConstructor
+public class GenericModifier implements DamageModifier {
+
+    private final String reason;
+    @With
+    private ModifierType type = ModifierType.GENERIC;
+    private final double multiplier;
+    private final double flat;
+
+    public GenericModifier(String reason, double multiplier, double flat) {
+        this.reason = reason;
+        this.multiplier = multiplier;
+        this.flat = flat;
+    }
```

Comment:
being able to specify both multiplier and flat could be problematic, would recommend just having an enum operator as to whether damage should be a flat increase/decrease or a multiplier